packages/worker/src/worker.ts

```diff
@@ -1492,6 +1492,29 @@ export class Worker {
     return takeUntil(this.stateSubject.pipe(filter((value) => value === state)));
   }
 
+  /**
+   * {@link run | Run} the Worker until `fnOrPromise` completes.
+   *
+   * @returns the result of `fnOrPromise`
+   *
+   * Throws on fatal Worker errors.
+   */
+  async runUntil<R>(fnOrPromise: Promise<R> | (() => Promise<R>)): Promise<R> {
```

Comment:
May be too late now, but I think we should make `shutdown()` async and not return until the entire shutdown is complete. That way, people could just `Promise.race([worker.run(), mySignalPromise]).finally(async () => await worker.shutdown())`.

Reply:
I think `runUntil` is cleaner, I clearly was missing it in all of the tests.

Reply:
As just a helper, sure (though I would like to avoid a slippery slope of _too_ many helpers).

But I still wonder if making shutdown async and having a non-async worker.start() could help. E.g.

```js
worker.start()
try {
  // Do stuff
} finally {
  await worker.shutdown()
}
```

Granted I could understand that people may not want to await shutdown but rather the original run, so maybe what you have here is best.